### PR TITLE
Hhd data gap fix

### PR DESCRIPTION
**Pull request by nickv23**
This is currently my fix for demogsurv functions that run calc_nqx and calc_nmx. In demog_pyears, there is a section where the data is run through pyears, where the output is an organized table stratified by strata + psu, sex_id (sometimes), admin_1_id (sometimes), and calculates the number of person-years and number of deaths for each stratification. In cases where none of the household deaths/ household members files contain any rows within the target age group, the pyears function returns an empty table, which can no longer be handled later on in the script. My solution is currently to repopulate the empty table with every stratified subgroup, and manually apply all calculated values to be 0. I set pyears to be 0.00001 because it later gets used as the denominator, and I preferred to get mx values of 0, rather than NaN. I can change this if there's a better way of handling this. The change to epis_labels function is to ensure that the age group ends and period ends are exclusive.

### CODE REVIEW

**Comment by hcomfo95** [comment]
Does the test prep run you launched yesterday include this change? Can you run a check to see whether this had any impact on person-years calculated for [CBH](https://github.com/ihmeuw-demographics/processing_gbd_mortality_RS/blob/msca_handoff/01_download_inputs.R#L1620-L1671) and [SIBS](https://github.com/ihmeuw-demographics/processing_gbd_mortality_RS/blob/msca_handoff/01_download_inputs.R#L1790-L1825)?

**Comment by nickv23** [comment]
> Does the test prep run you launched yesterday include this change? Can you run a check to see whether this had any impact on person-years calculated for [CBH](https://github.com/ihmeuw-demographics/processing_gbd_mortality_RS/blob/msca_handoff/01_download_inputs.R#L1620-L1671) and [SIBS](https://github.com/ihmeuw-demographics/processing_gbd_mortality_RS/blob/msca_handoff/01_download_inputs.R#L1790-L1825)?

Yes it does, although, I think I only changed the hh_collapse script to load the changes in, I will have to run another one where I load in this version of demogsurv in the survey collapse script as well to see how it affects it.

**Review by denny-uw — COMMENTED** [review]
I don't think this should affect CBH or SIBS, but it is good to double check. Does setting pyears to 0.00001 get you mx values of 0 or does it have decimals? Another solution to look at is the possibility of excluding the missing age group(s) from the demogsurv calculation so it doesn't return empty rows.

**Comment by nickv23** [comment]
> I don't think this should affect CBH or SIBS, but it is good to double check. Does setting pyears to 0.00001 get you mx values of 0 or does it have decimals? Another solution to look at is the possibility of excluding the missing age group(s) from the demogsurv calculation so it doesn't return empty rows.

Setting pyears to that small value gets mx values of 0, and if I were to change pyears to be 0, all mx values become NaN, and then they would have to be adjusted back to 0 later on.

File: R/demog_pyears.R [hunk 1]
```diff
@@ -2,7 +2,7 @@
   if("labels" %in% attributes(x))
     return(labels(x)[-length(x)])
   lower <- x[-length(x)]
-  upper <- x[-1]-1
+  upper <- x[-1]
```
Comment (erinamay) [inline R/demog_pyears.R:5]: was this change to fix a typo?

File: R/demog_pyears.R [hunk 2]
```diff
@@ -73,6 +74,30 @@ demog_pyears <- function(formula, data, period=NULL, agegr=NULL, cohort=NULL, ti
   })
   aggr <- Reduce(rbind, aggr_spl)
 
+  if (nrow(aggr) == 0) {
```
Comment (erinamay) [inline R/demog_pyears.R:77]: have you compared this with some of the changes Haley made recently to fill missing values outside of these functions in the survey script? Seems like we should choose one method or the other to fill values. @hcomfo95

**Comment by nickv23** [comment]
Closing this, not the ideal solution for this problem.

**Pull request closed without merging**